actionview/lib/action_view/helpers/form_tag_helper.rb

```diff
@@ -948,9 +948,14 @@ def set_default_disable_with(value, tag_options)
           tag_options.delete("data-disable-with")
         end
 
-        def convert_direct_upload_option_to_url(options)
+        def add_direct_upload_attributes(options)
           if options.delete(:direct_upload) && respond_to?(:rails_direct_uploads_url)
             options["data-direct-upload-url"] = rails_direct_uploads_url
+            if options[:direct_upload_model]
+              options["data-direct-upload-model"] = options.delete(:direct_upload_model)
+            elsif options[:object]
+              options["data-direct-upload-model"] = options[:object].class.name
```

Comment:
Could this be a signed reference (like [what Turbo Rails does for Stream names](https://github.com/hotwired/turbo-rails/blob/e9b5f68ff81184e2641c6faaedf13e9beedc71ff/app/channels/turbo/streams/stream_name.rb#L24-L29)) to prevent tampering?

Similarly, could the `Post` value be signed in the same way?

Reply:
> Could this be a signed reference (like [what Turbo Rails does for Stream names](https://github.com/hotwired/turbo-rails/blob/e9b5f68ff81184e2641c6faaedf13e9beedc71ff/app/channels/turbo/streams/stream_name.rb#L24-L29)) to prevent tampering?
> 
> Similarly, could the `model: Post` option be signed in the same way?

Yes, it could use `ActiveStorage.verifier`.

Could do something like:
```ruby
if options[:direct_upload_model_and_attribute] # Note change in name from direct_upload_model to direct_upload_model_and_attribute
  options["data-direct-upload-model-and-attribute"] = ActiveStorage.verifier.generate(options.delete(:direct_upload_model_and_attribute))
```